cmd/kube-apiserver/app/options/globalflags.go

```diff
@@ -0,0 +1,111 @@
+	"k8s.io/kubernetes/pkg/version/verflag"
+
+	// ensure libs have a chance to globally register their flags
+	_ "github.com/golang/glog"
+	_ "k8s.io/apiserver/pkg/admission"
+	_ "k8s.io/cloud-provider"
+)
+
+// AddGlobalFlags explicitly registers flags that libraries (glog, verflag, etc.)
+// register against the global flagsets from "flag" and "github.com/spf13/pflag".
+// We do this in order to prevent unwanted flags from leaking into the
+// kube-apiserver's flagset.
+func AddGlobalFlags(fs *pflag.FlagSet) {
+	addGlogFlags(fs)
+	verflag.AddFlags(fs)
+	logs.AddFlags(fs)
+}
+
+// AddGenericFlags explicitly registers flags that internal packages register
+// against the global flagsets from "flag". We do this in order to prevent
+// unwanted flags from leaking into the kube-apiserver's flagset.
+func AddGenericFlags(fs *pflag.FlagSet) {
+	addCloudProviderFlags(fs)
+	addAdmissionFlags(fs)
+}
+
+// normalize replaces underscores with hyphens. We should always use hyphens
+// instead of underscores when registering kube-apiserver flags.
+func normalize(s string) string {
+	return strings.Replace(s, "_", "-", -1)
+}
+
+// register adds a flag to local that targets the Value associated with the Flag
+// named globalName in global.
+func register(global *flag.FlagSet, local *pflag.FlagSet, globalName string) {
+	if f := global.Lookup(globalName); f != nil {
+		pflagFlag := pflag.PFlagFromGoFlag(f)
+		pflagFlag.Name = normalize(pflagFlag.Name)
+		local.AddFlag(pflagFlag)
+	} else {
+		panic(fmt.Sprintf("failed to find flag in global flagset (flag): %s", globalName))
+	}
+}
+
+// addCloudProviderFlags adds flags from k8s.io/kubernetes/pkg/cloudprovider.
+func addCloudProviderFlags(fs *pflag.FlagSet) {
+	// lookup flags in global flag set and re-register the values with our flagset
+	global := flag.CommandLine
+	local := pflag.NewFlagSet(os.Args[0], pflag.ExitOnError)
+
+	register(global, local, "cloud-provider-gce-lb-src-cidrs")
+
+	fs.AddFlagSet(local)
+}
+
+// addAdmissionFlags adds flags from k8s.io/kubernetes/plugin/pkg/admission.
+func addAdmissionFlags(fs *pflag.FlagSet) {
+	// lookup flags in global flag set and re-register the values with our flagset
+	global := flag.CommandLine
+	local := pflag.NewFlagSet(os.Args[0], pflag.ExitOnError)
+
+	register(global, local, "default-not-ready-toleration-seconds")
+	register(global, local, "default-unreachable-toleration-seconds")
+
+	fs.AddFlagSet(local)
+}
+
+// addGlogFlags adds flags from github.com/golang/glog.
+func addGlogFlags(fs *pflag.FlagSet) {
+	// lookup flags in global flag set and re-register the values with our flagset
+	global := flag.CommandLine
+	local := pflag.NewFlagSet(os.Args[0], pflag.ExitOnError)
+
+	register(global, local, "logtostderr")
+	register(global, local, "alsologtostderr")
+	register(global, local, "v")
+	register(global, local, "stderrthreshold")
+	register(global, local, "vmodule")
+	register(global, local, "log_backtrace_at")
+	register(global, local, "log_dir")
```

Comment:
do we have to register these by hardcoding their name here?

Reply:
I couldn't think of a better way since the flags are initialized the moment the `glog` package is used: 

https://github.com/kubernetes/kubernetes/blob/5336866a8c58a8fe943dff1746c1f179977b3aef/vendor/github.com/golang/glog/glog.go#L399-L404

We're doing the same thing in #68054 too. Previously without the named sections, we added everything from the global flag set: 

https://github.com/kubernetes/kubernetes/blob/5336866a8c58a8fe943dff1746c1f179977b3aef/cmd/kube-apiserver/apiserver.go#L47

With the named logical sections, we will need to extract out relevant flags so that we could group them.